features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/domain/ResourceRequestFactory.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.domain
+
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.net.Request
+import com.datadog.android.api.net.RequestFactory
+import com.datadog.android.api.storage.RawBatchEvent
+import com.datadog.android.sessionreplay.internal.exception.InvalidPayloadFormatException
+import okhttp3.RequestBody
+import okio.Buffer
+import java.util.Locale
+import java.util.UUID
+
+internal class ResourceRequestFactory(
+    internal val customEndpointUrl: String?,
+    private val resourceRequestBodyFactory: ResourceRequestBodyFactory = ResourceRequestBodyFactory()
+) : RequestFactory {
+
+    @Suppress("ThrowingInternalException")
+    override fun create(
+        context: DatadogContext,
+        batchData: List<RawBatchEvent>,
+        batchMetadata: ByteArray?
+    ): Request {
+        val applicationId = getApplicationId(context)
+        ?: throw InvalidPayloadFormatException(COULD_NOT_GET_APPLICATION_ID_ERROR)
+
+        val requestBody = resourceRequestBodyFactory
+            .create(applicationId, batchData)
+
+        return resolveRequest(context, requestBody)
+    }
+
+    private fun getApplicationId(datadogContext: DatadogContext): String? {
+        val rumContext = datadogContext.featuresContext[Feature.RUM_FEATURE_NAME]
```

Comment:
A random thought: should we store `application_id` when we write resource? Reading it at the upload cycle implies that RUM should be enabled at the time Session Replay is doing the upload and that application ID is the same as it was when resource was written.

Can it be the case that any of these statements can be false?

Not a blocker for me, just thinking.

Reply:
I've modified back to storing the applicationId along with the resource. The filename and application will then be properties of a JsonObject stored as the bytearray in rawBatchEvent metadata